ha/setup/create_pacemaker_resources.py

```diff
@@ -0,0 +1,251 @@
+op monitor interval=5s timeout=20s \
+op stop    interval=0s timeout=60s"
+    SimpleCommand().run_cmd(cmd)
+
+    if push:
+        cib_push(cib_xml)
+
+def csm_kibana(cib_xml, push=False):
+    """Creates mandatory resources for mgmt stack."""
+    kibana = f"pcs -f {cib_xml} resource create kibana systemd:kibana op monitor interval=30s"
+    agent = f"pcs -f {cib_xml} resource create csm-agent systemd:csm_agent op monitor interval=30s"
+    web = f"pcs -f {cib_xml} resource create csm-web systemd:csm_web op monitor interval=30s"
+
+    for c in (kibana, agent, web):
+        SimpleCommand().run_cmd(c)
+
+    if push:
+        cib_push(cib_xml)
+
+def uds(cib_xml, push=False):
+    """Create uds resource and constraints."""
+    cmd_uds = f"pcs -f {cib_xml} resource create uds systemd:uds op monitor interval=30s"
+    SimpleCommand().run_cmd(cmd_uds)
+    constraints = [
+            f"pcs -f {cib_xml} constraint colocation add uds with csm-agent score=INFINITY",
+            # According to EOS-9258, there is a bug which requires UDS to be started after csm_agent
+            f"pcs -f {cib_xml} constraint order csm-agent then uds"
+            ]
+    for c in constraints:
+        SimpleCommand().run_cmd(c)
+    if push:
+        cib_push(cib_xml)
+
+
+def mgmt_stack(cib_xml, kibana_vip_cfg, push=False):
+    """Wrapper to create Mgmt stack related resources.
+
+    It also creates and defines csm-kibana group to support colocation and
+    ordering requirements."""
+    csm_kibana(cib_xml)
+
+    uds(cib_xml)
+
+    cmd = f"pcs -f {cib_xml} resource group add csm-kibana kibana csm-agent csm-web"
+    SimpleCommand().run_cmd(cmd)
+
+    if kibana_vip_cfg:
+        kibana_vip(cib_xml, **kibana_vip_cfg, push=False)
+        cmd_group = f"pcs -f {cib_xml} resource group add csm-kibana kibana --before csm-agent"
+        SimpleCommand().run_cmd(cmd_group)
+
+    if push:
+        cib_push(cib_xml)
+
+
+def create_all_resources(cib_xml = "/tmp/cortx-lr2-cib.xml", push = True, **kwargs):
+    """Wrapper to populate the cluster.
+
+    Parameters:
+        cib_xml - file where CIB XML shall be stored (optional)
+        push - whether changes shall be actually applied (optional)
+        vip - kibana virtual ip (example: 10.0.0.6)
+        cidr - netmask for kibana vip (example: 24)
+        iface - interface to configure kibana vip (example: eno1)
+
+        vip, cidr, iface options are validated to be passed together.
+        ValueError exception is generated if set of argument is not empty but
+        incomplete.
+        """
+    kibana_vip_cfg = { k: kwargs[k] for k in ("vip", "cidr", "iface")
+                       if k in kwargs and kwargs[k] is not None }
+    if len(kibana_vip_cfg) not in (0, 3):
+        raise ValueError("Given kibana configuration is incomplete")
+
+    cib_get(cib_xml)
+
+    io_stack(cib_xml)
+
+    sspl(cib_xml)
+
+    mgmt_stack(cib_xml, kibana_vip_cfg)
+
+    if push:
+        cib_push(cib_xml)
```

Comment:
The cib_xml file, let's store it in log directory so that we know what has been pushed while debugging.

Reply:
Done.